Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -278,6 +290,54 @@ struct KlaviyoState: Equatable, Codable {
 
         return pushTokenData != newPushTokenData
     }
+
+    func buildProfileRequest(apiKey: String, anonymousId: String, properties: [String: Any] = [:]) -> KlaviyoAPI.KlaviyoRequest {
+        let payload = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.CreateProfilePayload(
+            data: .init(
+                profile: Profile(
+                    email: email,
+                    phoneNumber: phoneNumber,
+                    externalId: externalId,
+                    properties: properties),
+                anonymousId: anonymousId)
+        )
+        let endpoint = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.createProfile(payload)
+
+        return KlaviyoAPI.KlaviyoRequest(apiKey: apiKey, endpoint: endpoint)
+    }
+
+    mutating func buildTokenRequest(apiKey: String, anonymousId: String, pushToken: String, enablement: PushEnablement) -> KlaviyoAPI.KlaviyoRequest {
+        var profile: Profile
+
+        if let pendingProfile = pendingProfile {
+            profile = Profile.updateProfileWithProperties(
+                email: email,
+                phoneNumber: phoneNumber,
+                externalId: externalId,
+                dict: pendingProfile)
+            self.pendingProfile = nil
+        } else {
+            profile = Profile(email: email, phoneNumber: phoneNumber, externalId: externalId)
+        }
```

Comment:
This is the main change that fixes the issue. Earlier (before moving this code here from the state management file) we were not using the `pendingProfile` var from the state when making a token request and profile was constructed as it is in the else code block. `pendingProfile` is set whenever a user calls`KlaviyoSDK().set(profileAttribute: , value: )` and whenever we flush we see if it's nil and if not we enqueue a token or a profile request.

Also, since `pendingProfile` was never set to `nil` during a token request earlier, we kept making token request indefinitely whenever a profile property was set using the `setProfileAttribute` method. This is definitely a super edge case since most folks use the `set<Id>` methods or the `setProfile` method. This is now fixed by setting the ` self.pendingProfile = nil` on line 318.

Reply:
Thanks for this explanation, definitely not familiar enough with the iOS sdk code to have known this.